src/main/sensors/gyro_init.c

```diff
@@ -612,23 +605,24 @@ bool gyroInit(void)
         break;
     }
 
-    gyroDetectionFlags = GYRO_NONE_MASK;
+    gyroDetectionFlags = 0;
     uint8_t gyrosToScan = gyroConfig()->gyrosDetected;
 
-    gyro.gyroToUse = gyroConfig()->gyro_to_use;
+    gyro.gyroEnabledBitmask = gyroConfig()->gyro_enabled_bitmask;
     gyro.gyroDebugAxis = gyroConfig()->gyro_filter_debug_axis;
 
-    if ((!gyrosToScan || (gyrosToScan & GYRO_1_MASK)) && gyroDetectSensor(&gyro.gyroSensor1, gyroDeviceConfig(0))) {
-        gyroDetectionFlags |= GYRO_1_MASK;
-    }
-
-#if defined(USE_MULTI_GYRO)
-    if ((!gyrosToScan || (gyrosToScan & GYRO_2_MASK)) && gyroDetectSensor(&gyro.gyroSensor2, gyroDeviceConfig(1))) {
-        gyroDetectionFlags |= GYRO_2_MASK;
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        // Only attempt to detect a gyro if it's enabled or we're doing an auto-scan
+        if ((!gyrosToScan || (gyrosToScan & GYRO_MASK(i)) || (gyro.gyroEnabledBitmask & GYRO_MASK(i)))) {
```

Comment:
With bitmask handled above:
```suggestion
        if (gyrosToScan & GYRO_MASK(i)) {
```